### ISSUE TITLE
ResponseReturnTypeControllerActionRector does not work with FrameworkExtraBundle Template annotation/attribute

### ISSUE THREAD

**Issue by dbergomi** [rectorphp/rector-symfony#310]
### Starting code

    #[Route]
    #[Template]
    public function detail()
    {
        return ['param1' => 'param1Value'];
    }

### Current Behaviour
    #[Route]
    #[Template]
    public function detail()
    {
        return ['param1' => 'param1Value'];
    }

### Desired Behaviour
    #[Route]
    #[Template]
    public function detail(): ?array
    {
        return ['param1' => 'param1Value'];
    }

**Comment by johanadivare** [comment]
@dbergomi can you provide a minimal reproducible with https://getrector.org/demo?

**Comment by dbergomi** [comment]
Controller code

```
<?php

use Symfony\Bundle\FrameworkBundle\Controller\AbstractController;
use Symfony\Component\Routing\Annotation\Route;
use Symfony\Component\HttpFoundation\Response;
use Sensio\Bundle\FrameworkExtraBundle\Configuration\Template;

final class TestController extends AbstractController
{
    #[Route]
    #[Template]
    public function testDetail()
    {
        return [];
    }
}
```

Rector

```
<?php

use Rector\Config\RectorConfig;
use Rector\Set\ValueObject\SetList;
use Rector\Symfony\Rector\ClassMethod\ResponseReturnTypeControllerActionRector;

return static function (RectorConfig $rectorConfig): void {
    $rectorConfig->rule(ResponseReturnTypeControllerActionRector::class);
};
```

After the execution the code is unchanged by rector.

Result that would take into account the #[Template] attribute:

```
 final class TestController extends AbstractController
 {
     #[Route]
     #[Template]
 -   public function testDetail()
 +   public function testDetail(): ?array
     {
         return [];
     }
```

The function may also 'return null;' or not have a return at all if the template doens't need parameters or they are injected in the attribute #[Template]

**Comment by johanadivare** [comment]
Oh yeah https://getrector.org/demo/68176eff-de1f-4d95-b5ca-1fcd088f36b3 does not give output. Then we would need a pull request containing a test that fails. If this is what we want for behaviour

**Comment by stefantalen** [comment]
> Oh yeah https://getrector.org/demo/68176eff-de1f-4d95-b5ca-1fcd088f36b3 does not give output. Then we would need a pull request containing a test that fails. If this is what we want for behaviour

See https://github.com/rectorphp/rector-symfony/pull/311 ;)

**Comment by stefantalen** [comment]
@dbergomi This should be fixed with #311, can you confirm?

**Comment by TomasVotruba** [comment]
Yes, that's the case.

Closing then :+1:
